### PR TITLE
chore: improve the binding error message

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

* improve the error message for binder

Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - Covered by the exist tests.

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14144)

### CODE REVIEW

File: src/query/sql/src/planner/binder/bind_context.rs [hunk 1]
```diff
@@ -498,9 +499,10 @@ impl BindContext {
             metadata
                 .get_table_index(column_binding.database_name.as_deref(), table_name)
                 .ok_or_else(|| {
-                    ErrorCode::TableInfoError(format!(
-                        "Table `{table_name}` is not found in the metadata"
+                   ErrorCode::TableInfoError(format!(
```
Comment (sundy-li) [inline src/query/sql/src/planner/binder/bind_context.rs:502]: There will be two `:` now.

```
TableInfoError: Metadata Error:
```

File: src/query/sql/src/planner/binder/setting.rs [hunk 2]
```diff
@@ -58,15 +58,21 @@ impl Binder {
...
+                })?;
                 let vars = vec![VarValue {
                     is_global,
                     variable,
                     value,
                 }];
                 Ok(Plan::SetVariable(Box::new(SettingPlan { vars })))
             }
-            _ => Err(ErrorCode::SemanticError("value must be constant value")),
+            _ => Err(ErrorCode::SemanticError(
```
Comment (sundy-li) [inline src/query/sql/src/planner/binder/setting.rs:73]: ditto 
```
Semantic Error: Semantic error:
```

File: src/query/sql/src/planner/binder/bind_context.rs (hunk 1 above)
Comment (BohuTANG) [reply src/query/sql/src/planner/binder/bind_context.rs:502]: This one looks missing:)

**Comment by BohuTANG** [comment]
The binder error message is not same as others, it(SemanticError) not return to the client directly:
image

I closed this PR and refine them again while get a better way.

**Pull request closed without merging**